File: lib-esphome-client/src/main/java/io/github/arhor/esphome/client/async/internal/EspHomeConnection.java
package io.github.arhor.esphome.client.async.internal;

import io.github.arhor.esphome.client.async.model.EspHomeCommand;
import io.github.arhor.esphome.client.async.model.EspHomeEvent;

import java.util.concurrent.CompletableFuture;
import java.util.concurrent.Flow;

public interface EspHomeConnection extends AutoCloseable {

    CompletableFuture<Void> send(EspHomeCommand command);

    Flow.Publisher<EspHomeEvent> observeEvents();
}

File: lib-esphome-client/src/main/kotlin/io/github/arhor/esphome/client/EspHomeConnection.kt
package io.github.arhor.esphome.client

import io.github.arhor.esphome.client.model.EspHomeDeviceInfo
import io.github.arhor.esphome.client.model.EspHomeEntity

interface EspHomeConnection : AutoCloseable {
    fun deviceInfo(): EspHomeDeviceInfo
    fun fetchCameraImage(single: Boolean = true): ByteArray
    fun listEntities(): List<EspHomeEntity>
    fun subscribeStates(handler: EspHomeStateHandler)
}
